Objects/typeobject.c

```diff
@@ -2319,6 +2319,12 @@ valid_identifier(PyObject *s)
                      Py_TYPE(s)->tp_name);
         return 0;
     }
+    /* Since there is no way to return an error from PyUnicode_IsIdentifier()
+       we have to call PyUnicode_READY() to ensure that the string object is
+       in the "canonical" representation. */
```

Comment:
```suggestion
    /* Since there is no way to return an error from PyUnicode_IsIdentifier()
       we have to call explicitly PyUnicode_READY(). */
```